File: Task2/task2.cs
using System;
using System.Collections.Generic; 
public class Program
{
    public static void Main()
    {
        int numbers = Convert.ToInt32(Console.ReadLine());
        int max = 0;
        List<int> clumba = new List<int>(); 
        if (numbers <= 0)
        {
            Console.Write("Быть не может");
        }
        else
        {
            for (int i = 0; i < numbers; ++i)
            {
                int flower = Convert.ToInt32(Console.ReadLine());
                if (flower > 0)
                {
                    clumba.Add(flower);
                }
                if (flower > max)
                {
                    max = flower;
                }
            }
            int poliv = Convert.ToInt32(Console.ReadLine());
            if (poliv < max)
            {
                Console.Write("Как купишь новую лейку - приходи.");
            }
            else
            {
                int step = 0;
                int perem = poliv;
                for (int i = 0; i < clumba.Count; i++)
                {
                    if (perem < clumba[i])
                    {
                        step += (i * 2 + 1);
                        perem = poliv;
                    }
                    else
                    {
                        step += 1;
                    }
                    perem -= clumba[i];
                }
                Console.WriteLine(step);
            }
        }
    }
}

File: Task2/task2.php
<?php
$numbers = readline();
$max = 0;
if($numbers<=0) {
    echo "Быть не может";
    exit;
}
$clumba = []; 
for ($i = 0; $i < $numbers; $i++) {
    $flower = readline();
    if ($flower > 0) 
    {
         $clumba[] = $flower; 
    }
    if ($flower > $max)
    {
        $max = $flower;
    }
}
$poliv = readline();
if ($poliv < $max) {
    echo "Как купишь новую лейку - приходи.";
    exit;
}
$step = 0;
$perem = $poliv;
for ($i = 0; $i < count($clumba);$i++)
    {
        if ($perem < $clumba[$i])
        {
            $step = $step + ($i * 2 + 1);
            $perem = $poliv;
        }
        else
        {
            $step += 1;
        }
        $perem -= $clumba[$i];
    }
echo 'Результат: ' .$step;
?>
